### PR TITLE
Datasplash post

### PR DESCRIPTION
**Pull request by torbjornvatn**
Here's my stab at describing how the experience of using Datasplash in production has been.

### CODE REVIEW

File: _posts/2016-12-07-how-datasplash-improved-our-dataflow.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+This isn't very functional, but it's because of the underlying Java SDK.
+<code data-gist-id="89804fe22277ac79f5ca7ab22ebf7b71" data-gist-file="streaming_word_extract.clj" data-gist-line="81-82"/>
+
+Inside `apply-transforms-to-pipeline` we utilize the [Threading Macro](http://clojure.org/guides/threading_macros)
+to start passing the pipeline as the last argument to the `read-from-pubsub` transformation.
+The Threading Macro will then pass the result of that transformation as the last argument of the next one, and so on
+and so forth.
+<code data-gist-id="89804fe22277ac79f5ca7ab22ebf7b71" data-gist-file="streaming_word_extract.clj" data-gist-line="45,47"/>
+
+Here we se the actual processing of the data. For each message from PubSub we extract words (and flatten those lists
```
Comment (frodebjerke) [inline _posts/2016-12-07-how-datasplash-improved-our-dataflow.md]: se -> see

File: _posts/2016-12-07-how-datasplash-improved-our-dataflow.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+with mapcat), uppercase each word and add them to a simple row json object. Notice the different ways we pass functions
+to map/mapcat.
+<code data-gist-id="89804fe22277ac79f5ca7ab22ebf7b71" data-gist-file="streaming_word_extract.clj" data-gist-line="50,53,57"/>
+
+Last, but not least we write the results as separate lines to the given BigQuery table.
```
Comment (frodebjerke) [inline _posts/2016-12-07-how-datasplash-improved-our-dataflow.md]: lines -> rows

**Review by frodebjerke — APPROVED** [review]
Great great!

**Review by jarlefosen — APPROVED** [review]
LGTM 👍

File: _posts/2016-12-07-how-datasplash-improved-our-dataflow.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+## And in the blue corner, Datasplash!
+
+It's time to see of how Datasplash performs in the ring, and to showcase that I've chosen to reimplement the
+[*StreamingWordExtract*](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/blob/master/examples/src/main/java/com/google/cloud/dataflow/examples/complete/StreamingWordExtract.java)
+example from the Dataflow documentation. A Dataflow-off, so to speak.
+
+The example pipeline reads lines of text from a PubSub topic, splits each line into individual words, capitalizes those
+words, and writes the output to a BigQuery table
+
+Here's how it the code looks in it's [entirety](gist.github.com/torbjornvatn/89804fe22277ac79f5ca7ab22ebf7b71), and I'll
```
Comment (mastoj) [inline _posts/2016-12-07-how-datasplash-improved-our-dataflow.md]: broken link

File: _posts/2016-12-07-how-datasplash-improved-our-dataflow.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+Here we se the actual processing of the data. For each message from PubSub we extract words (and flatten those lists
+with mapcat), uppercase each word and add them to a simple row json object. Notice the different ways we pass functions
+to map/mapcat.
+<code data-gist-id="89804fe22277ac79f5ca7ab22ebf7b71" data-gist-file="streaming_word_extract.clj" data-gist-line="50,53,57"/>
+
+Last, but not least we write the results as separate lines to the given BigQuery table.
+<code data-gist-id="89804fe22277ac79f5ca7ab22ebf7b71" data-gist-file="streaming_word_extract.clj" data-gist-line="60-65"/>
+
+And thats it really! No
```
Comment (mastoj) [inline _posts/2016-12-07-how-datasplash-improved-our-dataflow.md]: thats -> that's

**Review by mastoj — COMMENTED** [review]
Great stuff.

File: _posts/2016-12-07-how-datasplash-improved-our-dataflow.md (hunk 3 above)
Comment (torbjornvatn) [reply _posts/2016-12-07-how-datasplash-improved-our-dataflow.md]: Good catch, probably a private gist

Comment (mastoj) [reply _posts/2016-12-07-how-datasplash-improved-our-dataflow.md]: I think it is public, but I think you need to add https://

**Comment by torbjornvatn** [comment]
Fixed the typos

**Comment by frodebjerke** [comment]
:shipit:

**Comment by larsbkrogvig** [comment]
screen shot 2016-12-08 at 14 11 10

Is the text _really_ supposed to decrease in size like this?

**Comment by jarlefosen** [comment]
@larsbkrogvig I was just assuming that was caused by the differences in Github Markdown processor and Jekyll. We're using Jekyll(?) on github-pages and I'm assuming @torbjornvatn has checked it locally that it doesn't get smaller and smaller

**Comment by larsbkrogvig** [comment]
As glitches go, it was a pretty funny one.

**Comment by torbjornvatn** [comment]
@larsbkrogvig no worries, it looks good in real (virtual) life

**Pull request merged**